albo.py

```diff
@@ -25,14 +26,33 @@ def main ():
 
     table = soup.find('div', id='boge')
 
-    new_id = int(table.find('tr').find_next_sibling().td.string)
+    top_row = table.find('tr').find_next_sibling()
+
+    if top_row is None:
+        print("Nessun annuncio disponibile")
+        return
+
+    new_id = int(top_row.td.string)
 
     headers = [header.string for header in table.find('tr').find_all("td")]
 
     ids_to_parse = cached_announcements + list(range(last_id + 1, new_id + 1))
 
+    #avoid channel flooding in case of error
+    if len(ids_to_parse) > 10 and '--force-unsafe' not in sys.argv:
+        ids_to_parse = []
+        print("Troppi annunci da parsare, usare --force-unsafe per forzare l'invio")
+
     for id in ids_to_parse:
-        tr = table.find('td', text=id).parent
+        td = table.find('td', text=id)
+
+        if (td is None):
+            tg_bot.send_debug_messages("ID non trovato: " + str(id) + "\n" + "Probabilmente lo hanno saltato")
+            if id in cached_announcements:
+                cached_announcements.remove(id)
+            continue
```

Comment:
???

Reply:
Sometimes it happens that a particular number is not used in sequential order (e.g. 934, 908), previously we just saved its id in the cached announcement file and kept searching for the aforementioned id, without any success. With this check the bot basically ignores that specific id number, sends a one-time debug message to the devs, in order to make sure it was not mistakenly discarded, and goes on as normal